src/main/java/org/rumbledb/runtime/functions/object/ObjectDescendantPairsFunctionIterator.java

```diff
@@ -120,4 +120,29 @@ private void getDescendantPairs(List<Item> items) {
             }
         }
     }
+
+    @Override
+    protected boolean hasNextLocal() {
+        return this.hasNext;
+    }
+
+    @Override
+    protected void resetLocal(DynamicContext context) {
+        this.iterator.open(this.currentDynamicContextForLocalExecution);
+        this.nextResults = new LinkedList<>();
```

Comment:
why not just clear the list rather than recreating one?

Reply:
Done.